Lib/fontbakery/profiles/googlefonts.py

```diff
@@ -3077,6 +3078,73 @@ def com_google_fonts_check_metadata_consistent_repo_urls(config, family_metadata
         yield PASS, "OK"
 
 
+@check(
+    id = 'com.google.fonts/check/metadata/primary_script',
+    conditions = ['family_metadata'],
+    rationale = """
+        Try to guess font's primary script and see if that's set in METADATA.pb.
+        This is an educated guess based on the number of glyps per script in the font
+        and should be taken with caution. This check is not 100% reliable.
+    """,
+    proposal = 'https://github.com/googlefonts/fontbakery/issues/4109'
+)
+def com_google_fonts_check_metadata_primary_script(ttFont, family_metadata):
+    """METADATA.pb: Check for primary_script"""
+    passed = True
+
+    def get_primary_script(ttFont):
+        from fontTools import unicodedata
+        from collections import Counter
+        script_count = Counter()
+        for x in ttFont.getBestCmap().keys():
+            for script in unicodedata.script_extension(chr(x)):
+                if script[0] != "Z":
```

Comment:
This skips `Zanb`, the code for Zanabazar square script. It would be safer to list the skipped codes like `Zinh` explicitly.

Reply:
![Screenshot from 2023-08-09 23-37-39](https://github.com/googlefonts/fontbakery/assets/213676/434eb611-e462-49b4-87ca-c13ab1b8be02)

I've now changed that line to only skip `Zinh`, `Zzzz` and `Zyyy`